pallets/contracts/src/lib.rs

```diff
@@ -369,7 +386,7 @@ decl_module! {
         /// * origin - Only owner of template is allowed to execute the dispatchable.
         /// * code_hash - Unique hash of the smart extension template.
         /// * new_url - New meta url that need to replace with old url.
-        #[weight = 400_000_000]
+        #[weight = <T as Trait>::WeightInfo::change_template_meta_url(new_url.as_ref().map_or(0, |u| u.0.len()) as u32 )]
```

Comment:
The weight doesn't depend on the URL length. It can be dropped from the benchmark.